types/tx/signing/signature.go

```
@@ -32,10 +32,15 @@
 func SignatureDataToProto(data SignatureData) *SignatureDescriptor_Data {
 	switch data := data.(type) {
 	case *SingleSignatureData:
+		signMode, err := APISignModeToInternal(data.SignMode)
+		if err != nil {
+			panic(fmt.Errorf("invalid sign mode: %v", err))
```

golangci-lint (1.62.2)

non-wrapping format verb for fmt.Errorf. Use `%w` to format errors

(errorlint)